Makefile

```diff
@@ -24,11 +25,15 @@ test:
 	go test -run=not-a-real-test ./...  # just ensures that the tests compile
 	go test ./...
 
-build-release:
-	$(GORELEASER_CMD) --snapshot --skip-publish --skip-validate
+$(GORELEASER):
```

Comment:
I'm not super familiar with `Makefile` syntax; what happens when you use `$()` syntax in the name of a target like this? And why would you want to do that here?

Reply:
1. What happens is just that it substitutes the variable, so the name of this target is really `./bin/goreleaser/goreleaser`.
2. That file won't exist until we download it. So having this target, and having the `build-release` and `publish-targets` depend on this target, is a way of saying "if you don't already have that file, here is how to get it."

Reply:
There are 2 basic kinds of makefile targets: ones that represent files, and ones that are targets you would explicitly run like `make build-release`. The latter are the ones that you'll see listed under `.PHONY`, which is a somewhat judgmental term make uses for "this isn't a file, so don't bother checking modification dates or anything like that if there happens to _also_ be a file called `build-release`".

Reply:
So, whenever target A has a dependency on target B (written as `A: B`), if B is not a file then make just goes ahead and runs B, but if B a file then make will check whether that file exists/is unchanged since the last build, and only runs the B commands if that is not the case.

Reply:
Cool! The filename-based makefile target type is definitely the piece I was missing; thanks!